shell/run_shell_cmd.go

```diff
@@ -139,22 +141,43 @@ func RunShellCommandWithOutput(
 			errWriter = jsonErrorWriter.WriterLevel(logrus.ErrorLevel)
 		}
 
-		var prefix = ""
-		if terragruntOptions.IncludeModulePrefix {
-			prefix = terragruntOptions.OutputPrefix
-		}
-
 		cmd.Dir = commandDir
 
 		// Inspired by https://blog.kowalczyk.info/article/wOYk/advanced-command-execution-in-go-with-osexec.html
-		cmdStderr := io.MultiWriter(withPrefix(errWriter, prefix), &stderrBuf)
+		cmdStderr := io.MultiWriter(log.TFStderrWriter(
+			errWriter,
+			terragruntOptions.Logger.Logger.Formatter,
+			terragruntOptions.OutputPrefix,
+			terragruntOptions.TerraformPath,
+		), &stderrBuf)
 
-		var cmdStdout io.Writer
+		cmdStdout := io.MultiWriter(&stdoutBuf)
 
 		if !suppressStdout {
-			cmdStdout = io.MultiWriter(withPrefix(outWriter, prefix), &stdoutBuf)
-		} else {
-			cmdStdout = io.MultiWriter(&stdoutBuf)
+			// display TF stdout as is (without wrapping to TG log) if args contains `-json` flag or `output` command is specified
+			var rawStdout bool
+
+			for i, arg := range args {
+				if (i == 0 && strings.EqualFold(arg, terraform.CommandNameOutput)) || strings.EqualFold(arg, terraform.FlagNameJSON) {
```

Comment:
NIT: This is a confusing line. I think extracting it into a well named function will make it easier to understand what we're doing here.